### PR TITLE
Generate a web session from an api key

### PR DESCRIPTION
**Pull request by acasajus**

### CODE REVIEW

File: app/auth/views/api_to_cookie.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from app.models import ApiToCookieToken
+from app.utils import sanitize_next_url
+
+
+@auth_bp.route("/api_to_cookie", methods=["POST"])
+def api_to_cookie():
+    if current_user.is_authenticated:
+        LOG.d("user is already authenticated, redirect to dashboard")
+        return redirect(url_for("dashboard.index"))
+    body = request.json
```
Comment (nguyenkims) [inline app/auth/views/api_to_cookie.py:16]: This is the url that app should open in the webview right? Do you know if the app can open a webview with a POST request? This URL is also used by the web extension and opening an URL with a POST request requires workaround like https://stackoverflow.com/a/17089124

File: app/models.py [hunk 2]
```diff
@@ -3296,3 +3296,12 @@ class NewsletterUser(Base, ModelMixin):
 
     user = orm.relationship(User)
     newsletter = orm.relationship(Newsletter)
+
+
+class ApiToCookieToken(Base, ModelMixin):
+
+    __tablename__ = "api_cookie_token"
+    code = sa.Column(sa.String(128), unique=True, nullable=False)
+    user_id = sa.Column(sa.ForeignKey(User.id, ondelete="cascade"), nullable=True)
```
Comment (nguyenkims) [inline app/models.py:3305]: can we add a foreign key to ApiKey to know which api key has been used to create a token?

Comment (acasajus) [reply app/models.py:3305]: Added

File: app/auth/views/api_to_cookie.py (hunk 1 above)
Comment (acasajus) [reply app/auth/views/api_to_cookie.py]: Ah. Good catch. I've changed it to a `GET` request :+1:

File: app/auth/views/api_to_cookie.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+def api_to_cookie():
+    if current_user.is_authenticated:
+        LOG.d("user is already authenticated, redirect to dashboard")
+        return redirect(url_for("dashboard.index"))
+    code = request.args.get("token")
+    if not code:
+        flash("Missing token", "error")
+        return redirect(url_for("auth.login"))
+
+    token = ApiToCookieToken.get_by(code=code)
```
Comment (nguyenkims) [inline app/auth/views/api_to_cookie.py:21]: Nits: I wonder if ApiToCookieToken.code should be renamed to ApiToCookieToken.token to be more consistent?

**Review by nguyenkims — APPROVED** [review]
LGTM, just left a nits :)

File: app/auth/views/api_to_cookie.py (hunk 3 above)
Comment (acasajus) [reply app/auth/views/api_to_cookie.py:21]: ApiKeys also use `code` in the db for the token. That's why I went for code here also.

**Pull request merged**